contract_manager/src/evm.ts

```diff
@@ -107,18 +146,87 @@ export class EVMContract extends Contract {
     return EVMContract.type;
   }
 
+  async getVersion(): Promise<string> {
+    const pythContract = this.getContract();
+    const result = await pythContract.methods.version().call();
+    return result;
+  }
+
+  static async deploy(
+    chain: EVMChain,
+    privateKey: string,
+    abi: any,
+    bytecode: string
+  ): Promise<EVMContract> {
+    const web3 = new Web3(chain.getRpcUrl());
+    const signer = web3.eth.accounts.privateKeyToAccount(privateKey);
+    web3.eth.accounts.wallet.add(signer);
+    const contract = new web3.eth.Contract(abi);
+    contract.options.data = bytecode;
+    const deployTx = contract.deploy({ data: "" });
```

Comment:
why we couldn't pass bytecode as data 🤔 can you comment that?

Reply:
I will try that 👍  This was a direct copy-paste from some example contract deployment code on the web